test/test_other.py

```diff
@@ -8631,29 +8631,45 @@ def test(infile, source_map_added_dir=''):
     test('inner/a.cpp', 'inner')
 
   def test_emsymbolizer(self):
-    # Test DWARF output
-    self.run_process([EMCC, test_file('core/test_dwarf.c'),
-                      '-g', '-O1', '-o', 'test_dwarf.js'])
+    def check_loc_info(address, source, funcs, locs):
+      out = self.run_process(
+          [emsymbolizer, '-tcode', '-s', source, 'test_dwarf.wasm', address],
+          stdout=PIPE).stdout
+      for func in funcs:
+        self.assertIn(func, out)
+      for loc in locs:
+        self.assertIn(loc, out)
 
     # Use hard-coded addresses. This is potentially brittle, but LLVM's
     # O1 output is pretty minimal so hopefully it won't break too much?
     # Another option would be to disassemble the binary to look for certain
     # instructions or code sequences.
 
-    def get_addr(address):
-      return self.run_process(
-          [emsymbolizer, 'test_dwarf.wasm', '-tcode', address], stdout=PIPE).stdout
-
-    # Check a location in foo(), not inlined.
-    # If the output binary size changes use `wasm-objdump -d` on the binary
-    # look for the offset of the first call to `out_to_js`.
-    self.assertIn('test_dwarf.c:6:3', get_addr('0x8'))
-    # Check that both bar (inlined) and main (inlinee) are in the output,
-    # as described by the DWARF.
-    # TODO: consider also checking the function names once the output format
-    # stabilizes more
-    self.assertRegex(get_addr('0x1f').replace('\n', ''),
-                     'test_dwarf.c:13:3.*test_dwarf.c:18:3')
+    # 1. Test DWARF + source map together
+    self.run_process([EMCC, test_file('core/test_dwarf.c'),
```

Comment:
I'm kind of on the fence about whether it would be good to try to reduce the repetition in these check_loc_info calls below. In general it's better to have more of this kind of repetition in tests than in production code (https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html if you haven't seen it) but given all these numbers it does seem like they could accidentally get out of sync.
Thoughts? I'm find if you want to leave it as-is.

Reply:
I checked the DAMP article now, but I'm not sure which repetition you want to restore..? We actually repeatedly use addresses and line numbers in the code, which can get out of sync easily as you suggested, but the article basically says "Don't try too hard to remove redundancies; listing things multiple times is actually better". Isn't that kind of what I did here? What `check_loc_info` does is to automate calling `emsymbolizer` and running assertions, not data.

Not that I agree or disagree with what you said; I'm just not sure which direction you suggest, for which I'd appreciate elaboration.

And one more thing... I think we can use `llvm-objdump` to remove the hardcoded addresses altogether in the next PR. After #17582 I think things will be pretty stable anyway, but as you suggested in https://github.com/emscripten-core/emscripten/pull/17582#issuecomment-1206809870, I think automating the code offset calculation is still worth it.

Reply:
Yes, you did basically what the article is suggesting. I'm just looking at all that repetition and thinking... "do we *really* want that much?" 🤣 
But ultimately I stopped short of actually suggesting that you change it.
And in any case if you want to use `objdump` then there will be less of it anyway. So probably best to leave it as it is.

Reply:
Ah sorry now I read your comments again you were not suggesting I follow the article but rather like "we have this principle but isn't it too much?... Yeah I'll try to reduce the repetition in the follow-up PR.